File: packages/python-sdk/src/certifieddata_payments/resources/events.py
"""Events resource."""

from typing import Any, Optional
from .._http import HttpClient


class EventsResource:
    def __init__(self, http: HttpClient) -> None:
        self._http = http

    def get(self, event_id: str) -> dict[str, Any]:
        return self._http.request("GET", f"/v1/events/{event_id}")

    def list(
        self,
        *,
        event_type: Optional[str] = None,
        limit: Optional[int] = None,
        starting_after: Optional[str] = None,
    ) -> dict[str, Any]:
        return self._http.request(
            "GET",
            "/v1/events",
            params={
                "event_type": event_type,
                "limit": limit,
                "starting_after": starting_after,
            },
        )

File: packages/typescript-sdk/src/resources/events.ts
import type { CDPClientConfig } from "../utils/request.js";
import { makeRequest } from "../utils/request.js";
import type { PaginationEnvelope, ListParams } from "../types/common.js";
import type { Event } from "../types/resources.js";

export class EventsResource {
  constructor(private readonly config: CDPClientConfig) {}

  async get(eventId: string): Promise<Event> {
    return makeRequest(this.config, "GET", `/v1/events/${eventId}`);
  }

  async list(params?: ListParams & { event_type?: string }): Promise<PaginationEnvelope<Event>> {
    return makeRequest(this.config, "GET", "/v1/events", {
      query: {
        limit: params?.limit,
        starting_after: params?.starting_after,
        event_type: params?.event_type,
      },
    });
  }
}
